simperby/src/types.rs

```diff
@@ -0,0 +1,65 @@
+use super::*;
+use serde::{Deserialize, Serialize};
+use simperby_repository::raw::SemanticCommit;
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct ConsensusStatus {
+    // TODO
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct NetworkStatus {
+    // TODO
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub enum CommitInfo {
+    Block {
+        semantic_commit: SemanticCommit,
+        block_header: BlockHeader,
+        // TODO: block-specific consensus status
+    },
+    Agenda {
+        semantic_commit: SemanticCommit,
+        agenda: Agenda,
+        voters: Vec<(MemberName, Timestamp)>,
+    },
+    AgendaProof {
+        semantic_commit: SemanticCommit,
+        agenda_proof: AgendaProof,
+    },
+    Transaction {
+        semantic_commit: SemanticCommit,
+        transaction: Transaction,
+    },
+    PreGenesisCommit {
```

Comment:
What is this for??

Reply:
Literally, commits before the genesis commit. FYI, this has existed before this PR (I just moved the location)